consensus/src/pipeline/virtual_processor/processor.rs

```diff
@@ -480,6 +481,21 @@ impl VirtualStateProcessor {
         (virtual_parents, ghostdag_data)
     }
 
+    pub fn validate_transaction_and_populate(&self, mutable_tx: &mut MutableTransaction) -> Result<(), RuleError> {
+        let virtual_utxo_view = self.virtual_utxo_store.as_ref(); // TODO: manage lock with virtual state lock
+        let virtual_state = self.virtual_state_store.read().get().unwrap(); // TODO: lock
+        let virtual_daa_score = virtual_state.daa_score;
+        let virtual_past_median_time = self.past_median_time_manager.calc_past_median_time(&virtual_state.ghostdag_data).0; // TODO: save median time in virtual state
+
+        self.transaction_validator.validate_tx_in_isolation(&mutable_tx.tx).unwrap(); // TODO: return rule error
+        self.transaction_validator.utxo_free_tx_validation(&mutable_tx.tx, virtual_daa_score, virtual_past_median_time).unwrap(); // TODO: return rule error
+        let validated_tx = self.validate_transaction_in_utxo_context(&mutable_tx.tx, virtual_utxo_view, virtual_daa_score).unwrap(); // TODO: return rule error
```

Comment:
This behaviour of `validate_transaction_in_utxo_context` does not seem to match glolang impl and does not entirely fulfill mempool needs.

We want to fill `mutable_tx` with all available entries even if a `ValidatedTransaction` cannot be built because some entries are still missing.

Your intention to introduce  `Vec<Option<UtxoEntry>>` is a perfect fit for this.

We also need to populate `calculated_mass`. Just wondering if this could it be delegated to transaction_validator_populated.rs `validate_populated_transaction_and_get_fee`. Wdyt?